features/support/steps.js

```diff
@@ -17,19 +17,21 @@ Given(/we have a function that can add two numbers called (.*)/, (functionName,
 });
 
 Given(/we want to use (.*) as the (.*) number/, (number, place, done) => {
-  world[place] = parseInt(number, 10);
+  world.params[place] = parseInt(number, 10);
   done();
 });
 
 When(/we add our numbers using the (.*) function/, (functionName, done) => {
-  expect(world.first).to.be.a(`number`);
-  expect(world.second).to.be.a(`number`);
+  const list = Object.values(world.params);
+  list.forEach( place => {
+    expect(place).to.be.a(`number`);
+  });
   expect(ourFunctions[functionName]).to.be.a(`function`);
-  world.result = ourFunctions[functionName](world.first, world.second);
+  world.result = (ourFunctions[functionName]).apply(null, list);
```

Comment:
* I like this!  At first we said there should only be 2 arguments but making it so that there can be any number of arguments makes it more interesting...